lib/ansible/module_utils/basic.py

```diff
@@ -1817,6 +1823,7 @@ def _handle_options(self, argument_spec=None, params=None, prefix=''):
 
                     options_legal_inputs = list(spec.keys()) + list(options_aliases.keys())
 
+                    self._set_internal_properties(spec, param)
```

Comment:
Per our discussion, I think this will be fine to duplicate the call here. If it were contained within `_check_arguments` itself, we'd guarantee that we didn't accidentally call one without the other, and we could make the args to `_set_internal_properties` non-optional, but I also don't see us needing to add any more calls to `_check_arguments` either. So either way you prefer I suppose.

Reply:
I like having them separate as separate calls. It keeps the methods focused rather than trying to do too many things.